### ISSUE TITLE
K8S ConfigMap doesn't work very well with profile

### ISSUE THREAD

**Issue by cmoulliard** [spring-cloud-incubator/spring-cloud-kubernetes#106]
When the following configMap is defined 

```
metadata:
  name: store
data:
  application.yml: |-
    greeting:
      message: Say Hello to the World
    ---
    spring:
      profiles: development
    greeting:
      message: Say Hello to the Developers
    ---
    spring:
      profiles: production
    greeting:
      message: Say Hello to the Ops
```

and the profile `development` specified

```
exec java -Dspring.profiles.active=development -javaagent:/opt/jolokia/jolokia.jar=config=/opt/jolokia/etc/jolokia.properties -cp . -jar /deployments/store-1.0-SNAPSHOT.jar
I> No access restrictor found, access to any MBean is allowed
Jolokia: Agent started with URL https://172.17.0.13:8778/jolokia/
14:40:35.904 [background-preinit] INFO  o.h.validator.internal.util.Version - HV000001: Hibernate Validator 5.3.5.Final
14:40:36.195 [main] INFO  o.s.c.a.AnnotationConfigApplicationContext - Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@7e0babb1: startup date [Thu Jun 22 14:40:36 UTC 2017]; root of context hierarchy
14:40:36.524 [main] INFO  o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'configurationPropertiesRebinderAutoConfiguration' of type [org.springframework.cloud.autoconfigure.ConfigurationPropertiesRebinderAutoConfiguration$$EnhancerBySpringCGLIB$$f55351cc] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v1.5.3.RELEASE)
14:40:37.000 [main] INFO  o.s.c.b.c.PropertySourceBootstrapConfiguration - Located property source: ConfigMapPropertySource [name='configmap.store.jbcnf']
14:40:37.004 [main] INFO  o.s.c.b.c.PropertySourceBootstrapConfiguration - Located property source: SecretsPropertySource [name='secrets.store.jbcnf']
14:40:37.068 [main] INFO  my.cool.store.HelloWorld - The following profiles are active: development,kubernetes
```

the rest endpoint defined within the controller

```
@EnableConfigurationProperties
@ConfigurationProperties("greeting")
@RestController
public class SayController {

	public String getMessage() {
		return Message;
	}

	public void setMessage(String message) {
		Message = message;
	}

	public String Message;

	@RequestMapping("/say")
	public String sayHello() {
		return "Message : " + Message;
	}
}

```

returns the wrong message

```
http http://store-jbcnf.192.168.64.25.nip.io/say
HTTP/1.1 200
Cache-control: private
Content-Length: 30
Content-Type: text/plain;charset=UTF-8
Date: Thu, 22 Jun 2017 14:45:54 GMT
Set-Cookie: 81040242305c5dd1b337c37d137087e7=91b529ca60b772928dda9f65023ba587; path=/; HttpOnly

Message : Say Hello to the Ops
```

It seems that only the latest key is loaded ...

Keys/values are well parsed 

```
18:33:35.519 [main] DEBUG o.s.b.f.c.YamlPropertiesFactoryBean - Loading from YAML: Byte array resource [resource loaded from byte array]
18:33:35.520 [main] DEBUG o.s.b.f.c.YamlPropertiesFactoryBean - Merging document (no matchers set): {greeting={message=Say Hello to the World}}
18:33:35.521 [main] DEBUG o.s.b.f.c.YamlPropertiesFactoryBean - Merging document (no matchers set): {spring={profiles=development}, greeting={message=Say Hello to the Developers}}
18:33:35.522 [main] DEBUG o.s.b.f.c.YamlPropertiesFactoryBean - Merging document (no matchers set): {spring={profiles=production}, greeting={message=Say Hello to the Ops}}
18:33:35.522 [main] DEBUG o.s.b.f.c.YamlPropertiesFactoryBean - Loaded 3 documents from YAML resource: Byte array resource [resource loaded from byte array]
18:33:35.533 [main] INFO  o.s.c.b.c.PropertySourceBootstrapConfiguration - Located property source: ConfigMapPropertySource [name='configmap.store.jbcnf']
```

The existing code should take care of the profile as the class `org.springframework.boot.env.YamlPropertySourceLoader` of spring Boot does

```
/**
	 * {@link YamlProcessor} to create a {@link Map} containing the property values.
	 * Similar to {@link YamlPropertiesFactoryBean} but retains the order of entries.
	 */
	private static class Processor extends YamlProcessor {

		Processor(Resource resource, String profile) {
			if (profile == null) {
				setMatchDefault(true);
				setDocumentMatchers(new SpringProfileDocumentMatcher());
			}
			else {
				setMatchDefault(false);
				setDocumentMatchers(new SpringProfileDocumentMatcher(profile));
			}
```

**Comment by cmoulliard** [comment]
Fix #107

**Comment by geoand** [comment]
This seems to be an issue again (most likely due to the refactoring work done for Spring Boot 2.x)
